src/timesync/timesyncd-manager.c

```diff
@@ -127,13 +131,33 @@ static int manager_send_request(Manager *m) {
 
         server_address_pretty(m->current_server_address, &pretty);
 
-        len = sendto(m->server_socket, &ntpmsg, sizeof(ntpmsg), MSG_DONTWAIT, &m->current_server_address->sockaddr.sa, m->current_server_address->socklen);
-        if (len == sizeof(ntpmsg)) {
-                m->pending = true;
-                log_debug("Sent NTP request to %s (%s).", strna(pretty), m->current_server_name->string);
+        if (m->ntske) {
+#if GNUTLS_FOR_NTS
+                size_t packet_size;
+
+                r = ntp_extension_build_request_packet(m, &ntpmsg, &packet_size);
+                if (r < 0)
+                        return r;
+
+                len = sendto(m->server_socket, &ntpmsg, packet_size, MSG_DONTWAIT, &m->current_server_address->sockaddr.sa, m->current_server_address->socklen);
+                if (len == (int) packet_size) {
+                        m->pending = true;
+                        log_debug("Sent NTS request to %s (%s).", strna(pretty), m->current_server_name->string);
+                } else {
+                        log_debug_errno(errno, "Sending NTS request to %s (%s) failed: %m", strna(pretty), m->current_server_name->string);
+                        return manager_connect(m);
+                }
+#endif
```

Comment:
In this \#ifdef and more in the common files, shouldn't there be \#else branch to handle those cases? In other words, while the code now compiles without the required newer version of gnutls, what happens when running it?